src/util/types.ts

```diff
@@ -1,45 +1,65 @@
 import * as semver from "semver";
-
 import { Inputs } from "../generated/inputs-outputs";
 
-// https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#template-literal-types
-// neat
-export type InstallableClient = `${Inputs}`;
+// https://www.typescriptlang.org/docs/handbook/utility-types.html#excludetype-excludedunion
+export type InstallableClient = Exclude<`${Inputs}`, "source" | "github_pat">;
 
 export type ClientsToInstall = { [key in InstallableClient]?: semver.Range };
 
 /**
  * Store here any details for the client that do not match the "expected" values.
+ * GithubRepositoryPath has the repository path of the respective client
  * For example, directoryName usually matches the executable name (the InstallableClient),
  * but if it doesn't, it's overridden here.
+ * isHashMissingOn* stores boolean value that indicates if hash file is missing or not in mirror/github
  */
 export const ClientDetailOverrides: { [key in InstallableClient]?: {
-    directoryName?: string;
+    mirrorDirectoryName?: string;
+    githubRepositoryPath?: string;
+    isHashMissingOnGithub?: boolean;
+    isHashMissingOnMirror?: boolean;
     // executableName?: string;
 }} = {
     kam: {
-        directoryName: "kam",
+        mirrorDirectoryName: "kam",
+        githubRepositoryPath: "redhat-developer/kam",
+        isHashMissingOnGithub: true,
     },
     kamel: {
-        directoryName: "camel-k",
+        mirrorDirectoryName: "camel-k",
+        githubRepositoryPath: "apache/camel-k",
+        isHashMissingOnGithub: true,
     },
     kn: {
-        directoryName: "serverless",
+        mirrorDirectoryName: "serverless",
+        // There is no stable release present here https://github.com/knative/client/releases as of now.
```

Comment:
we should stlil add the prerelease kn client, I don't see them going into stable anytime soon if they haven't already.

Reply:
Okay :+1: